server/src/main/java/com/genymobile/scrcpy/SurfaceEncoder.java

```diff
@@ -0,0 +1,288 @@
+package com.genymobile.scrcpy;

```

Comment:
You can review this file by comparing it to (the original) `ScreenEncoder.java`. It didn't change much except extracting some logics to abstract methods.